ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js

```diff
@@ -0,0 +1,151 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+
+(function() {
+    "use strict";
+    const DC_TITLE = 'dc:title';
+    const REPO_PATH = 'repo:path';
+    const FIELD_TYPE = 'fieldType';
+    const PARENT_ID = 'parentId';
+    
+    let dataLayer;
+    let formTitle, fieldTitle, fieldType, panelTitle, formContainer;
+    let startTime = (new Date()).getTime();
+
+    const FASTTRACK_ANALYTICS_EVENT = "FormFastTrackAnalyticsEvent";
+    const FormEvents = {
+        RENDER: "Form renditions",
+        SUBMIT: "Form submissions",
+        ERROR: "Validation errors",
+        FIELD: "Field visits",
+        ABANDON: "Abandoned forms",
+        HELP: "Help views",
+        TIMESPENT: "Time Spent"
+    }
+
+    const getFormName = (dataLayerContent) => {
+        formContainer = Object.keys(dataLayerContent.page)[0];
+        const formName = dataLayerContent.page[formContainer][DC_TITLE];
+        let formPath = dataLayerContent.page[formContainer][REPO_PATH];
+        formPath = formPath.substring(0,formPath.lastIndexOf('/'));
+        return formName + '(' + formPath + ')';
+    }
+    
+    const getCurrentFieldData = (fieldId, dataLayerContent) => {
+        return dataLayerContent.component[fieldId];
+    }
+    
+    const getFieldPanelTitle = (fieldData, dataLayerContent) => {
+        if(dataLayerContent.component[fieldData[PARENT_ID]] === undefined) {
+            return dataLayerContent.page[fieldData[PARENT_ID]][DC_TITLE];
+        }
+        return dataLayerContent.component[fieldData[PARENT_ID]][DC_TITLE] ;
+    }
+
+    const getEventInfo = (_eventTitle, _formTitle, _fieldId, dataLayerContent) => {
+        // global variables to be used in abandon event which can be triggered anytime
+        const fieldData = getCurrentFieldData(_fieldId, dataLayerContent);
+        formTitle = _formTitle;
+        fieldTitle = fieldData[DC_TITLE];
+        fieldType = fieldData[FIELD_TYPE];
+        panelTitle = getFieldPanelTitle(fieldData, dataLayerContent);
+        return {
+            formTitle,
+            fieldTitle,
+            fieldType,
+            panelTitle,
+            eventName: _eventTitle
+        }
+    }
+
+    const registerGuideBridgeEvents = () => {
+        window.guideBridge.connect(()=>{
+            const dataLayerContent = dataLayer.getState();
+            formTitle = getFormName(dataLayerContent);
```

Comment:
we already support form title in the Form schema, so you need not derive it.

Reply:
removed custom implementation of formTitle